libparsec/crates/testbed/src/env.rs

```diff
@@ -277,6 +277,130 @@ impl TestbedEnv {
+                _ => None,
+            })
+            .unwrap()
+    }
+
+    pub fn get_device_certificate(
+        &self,
+        device_id: impl TryInto<DeviceID>,
+    ) -> (Arc<DeviceCertificate>, Bytes) {
+        let device_id = device_id
+            .try_into()
+            .unwrap_or_else(|_| panic!("Invalid DeviceID !"));
+
+        self.template
+            .certificates()
+            .find_map(|event| match event.certificate {
+                AnyArcCertificate::Device(certif) if certif.device_id == device_id => {
+                    Some((certif, event.signed))
+                }
+                _ => None,
+            })
+            .unwrap()
+    }
+
+    pub fn get_revoked_certificate(
+        &self,
+        user_id: impl TryInto<UserID>,
+    ) -> (Arc<RevokedUserCertificate>, Bytes) {
+        let user_id = user_id
+            .try_into()
+            .unwrap_or_else(|_| panic!("Invalid UserID !"));
+
+        self.template
+            .certificates()
+            .find_map(|event| match event.certificate {
+                AnyArcCertificate::RevokedUser(certif) if certif.user_id == user_id => {
+                    Some((certif, event.signed))
+                }
+                _ => None,
+            })
+            .unwrap()
+    }
+
+    pub fn get_realm_role_certificate(
+        &self,
+        user_id: impl TryInto<UserID>,
+    ) -> (Arc<RealmRoleCertificate>, Bytes) {
+        let user_id = user_id
+            .try_into()
+            .unwrap_or_else(|_| panic!("Invalid UserID !"));
+
+        self.template
+            .certificates()
+            .find_map(|event| match event.certificate {
+                AnyArcCertificate::RealmRole(certif) if certif.user_id == user_id => {
+                    Some((certif, event.signed))
+                }
+                _ => None,
+            })
+            .unwrap()
+    }
+
+    pub fn get_user_update_certificate(
+        &self,
+        user_id: impl TryInto<UserID>,
+    ) -> (Arc<UserUpdateCertificate>, Bytes) {
+        let user_id = user_id
+            .try_into()
+            .unwrap_or_else(|_| panic!("Invalid UserID !"));
+
+        self.template
+            .certificates()
+            .find_map(|event| match event.certificate {
+                AnyArcCertificate::UserUpdate(certif) if certif.user_id == user_id => {
+                    Some((certif, event.signed))
+                }
+                _ => None,
+            })
+            .unwrap()
+    }
+
+    pub fn get_sequester_authority_certificate(
+        &self,
+    ) -> (Arc<SequesterAuthorityCertificate>, Bytes) {
+        self.template
+            .certificates()
+            .find_map(|event| match event.certificate {
+                AnyArcCertificate::SequesterAuthority(certif) => Some((certif, event.signed)),
+                _ => None,
+            })
+            .unwrap()
+    }
+
+    pub fn get_sequester_service_certificate(
+        &self,
+        index: usize,
```

Comment:
use `service_id: SequesterServiceID` to identify the certificate

Reply:
Done